### PR TITLE
*: prepare v3 branch

### PR DESCRIPTION
**Pull request by simonpasquier**
github.com/simonpasquier/klog-gokit/v3 will only support k8s.io/klog/v2
with github.com/go-kit/kit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,7 +39,7 @@ logger = level.NewFilter(logger, level.AllowAll())
 klog.SetLogger(logger)
 ```
 
-Setting the logger to the klog package **MUST** happen before using klog in any package.
+Setting the klog's logger **MUST** happen at the very beginning of your program (e.g. before using the other klog functions).
```
Comment (LeviHarrison) [inline README.md]: nit: This sounds a little weird

```suggestion
Setting klog's logger **MUST** happen at the very beginning of your program (e.g. before using the other klog functions).
```

File: klog.go [hunk 2]
```diff
@@ -54,11 +56,23 @@ func (v Verbose) Infof(format string, args ...interface{}) {
...
+		level.Debug(logger).Log("func", "Verbose.InfoS", "msg", msg, keysAndValues)
+	}
+}
+
+func (v Verbose) InfoSDepth(_ int, msg string, keysAndValues ...interface{}) {
+	if v {
+		level.Debug(logger).Log("func", "Verbose.InfoSDepth", "msg", msg, keysAndValues)
+	}
+}
+
```
Comment (LeviHarrison) [inline klog.go:70]: Do these new functions need to be added to the table in the README?

File: klog.go [hunk 3]
```diff
@@ -36,6 +36,8 @@ type Verbose bool
 
 func V(level Level) Verbose { return level <= maxLevel }
 
+func (v Verbose) Enabled() bool { return bool(v) }
```
Comment (LeviHarrison) [inline klog.go:39]: Also this one... maybe?

**Review by LeviHarrison — COMMENTED** [review]
Thanks so much for all your help with this!

File: klog.go (hunk 2 above)
Comment (simonpasquier) [reply klog.go:70]: nice catch!

File: klog.go (hunk 3 above)
Comment (simonpasquier) [reply klog.go:39]: The table helps people understand the mapping between go-kit and klog levels. `Enabled()` only exists to match the klog interface but has no direct mapping in go-kit so it's not worth mentioning it in the docs IMHO.

**Comment by simonpasquier** [comment]
@LeviHarrison I've just pushed a new version to address your comments!

**Review by LeviHarrison — APPROVED** [review]
LGTM!

**Pull request merged**